pymc3/distributions/multivariate.py

```diff
@@ -2048,3 +2049,51 @@ def random(self, point=None, size=None):
 
     def _distr_parameters_for_repr(self):
         return ["mu", "W", "alpha", "tau"]
+
+
+class ICARRV(RandomVariable):
+    name = "icar"
+    ndim_supp = 1
+    ndims_params = [2, 0]
+    dtype = "floatX"
+    _print_name = ("ICAR", "\\operatorname{ICAR}")
+
+    @classmethod
+    def rng_fn(cls, rng, A, tau, size):
+        n = A.shape[0]
+        mu = np.zeros(n)
+        if A.ndim != 2 or n != A.shape[1] or np.linalg.norm(A - A.T, np.inf) > 1e-06:
+            raise ValueError("The adjacency matrix `A` needs to be a square and symmetric")
+        W = np.diag(A.sum(axis=0)) - A
+        cov = np.linalg.pinv(W) / tau
+        return multivariate_normal.rng_fn(rng=rng, mean=mu, cov=cov, size=size)
+
+
+icar = ICARRV()
+
+
+class ICAR(Continuous):
+    r"""
+    The zero-sum constrained Intrinsic Conditional Autoregressive Model.
```

Comment:
These docstrings need to be expanded. Also the distribution has to be included in the docs.

Reply:
I plan to expand on it as soon as I get everything working.